### PR TITLE
Maintenance | New methods definition syntax

### PR DESCRIPTION
**Pull request by bkoval**
As stated in the guidelines repository ( https://github.com/Wikia/guidelines/tree/master/TypeScript#object-method-shorthand ) we should be using shorthand for defining object literal methods everywhere.

### CODE REVIEW

File: front/scripts/main/components/ForumWrapperComponent.ts [hunk 1]
```diff
@@ -12,12 +12,12 @@ App.ForumWrapperComponent = Em.Component.extend(App.LoadingSpinnerMixin, {
...
 
 	actions: {
-		goToPost: function (postId: number): void {
+		goToPost(postId: number): void {
 			this.sendAction('goToPost', postId);
 		}
 	},
 
-	didScroll: function() {
+	didScroll() {
```
Comment (Warkot) [inline front/scripts/main/components/ForumWrapperComponent.ts]: Missing return type.

File: front/scripts/main/components/ForumWrapperComponent.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ App.ForumWrapperComponent = Em.Component.extend(App.LoadingSpinnerMixin, {
 		}
 	},
 
-	hasMore: function() {
+	hasMore() {
```
Comment (Warkot) [inline front/scripts/main/components/ForumWrapperComponent.ts]: Missing return type.

File: front/scripts/main/components/ForumWrapperComponent.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ App.ForumWrapperComponent = Em.Component.extend(App.LoadingSpinnerMixin, {
 	}),
 
 	// Check if scrolling should trigger fetching new posts
-	isScrolledToTrigger: function() {
+	isScrolledToTrigger() {
```
Comment (Warkot) [inline front/scripts/main/components/ForumWrapperComponent.ts]: Missing return type.

File: front/scripts/main/components/LightboxWrapperComponent.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ App.LightboxWrapperComponent = Em.Component.extend({
 		return type ?  'lightbox-' + type : null;
 	}),
 
-	click: function (event: MouseEvent) {
+	click(event: MouseEvent) {
```
Comment (Warkot) [inline front/scripts/main/components/LightboxWrapperComponent.ts]: Missing return type.

File: front/scripts/main/components/PortableInfoboxComponent.ts [hunk 5]
```diff
@@ -85,7 +85,7 @@ App.PortableInfoboxComponent = Em.Component.extend(App.ArticleContentMixin, App.
 	 * @desc In case of long infobox, setups click
 	 * handling function to this infobox component.
 	 */
-	didInsertElement: function () {
+	didInsertElement() {
```
Comment (Warkot) [inline front/scripts/main/components/PortableInfoboxComponent.ts]: Missing return type.

File: front/scripts/main/components/SideNavComponent.ts [hunk 6]
```diff
@@ -50,7 +50,7 @@ App.SideNavComponent = Em.Component.extend({
 		 * Temporary solution for enter on search, will be refactored to be a route in mercury
 		 * @param value of input
 		 */
-		enter: function (value = '') {
+		enter(value = '') {
```
Comment (Warkot) [inline front/scripts/main/components/SideNavComponent.ts]: Missing return type.

File: front/scripts/main/components/UserStatusComponent.ts [hunk 7]
```diff
@@ -86,7 +86,7 @@ App.UserStatusComponent = Em.Component.extend({
 	 * Add redirect URL on click to make sure it accurately reflects current URL
 	 * @param event
 	 */
-	click: function (event: Event) {
+	click(event: Event) {
```
Comment (Warkot) [inline front/scripts/main/components/UserStatusComponent.ts]: Missing return type.

File: front/scripts/main/components/WikiaFooterComponent.ts [hunk 8]
```diff
@@ -52,7 +52,7 @@ App.WikiaFooterComponent = Em.Component.extend(App.TrackClickMixin, {
 		}
 	],
 	actions: {
-		handleFooterLinkClick: function (text: string, href: string) {
+		handleFooterLinkClick(text: string, href: string) {
```
Comment (Warkot) [inline front/scripts/main/components/WikiaFooterComponent.ts]: Missing return type.

File: front/scripts/main/controllers/ApplicationController.ts [hunk 9]
```diff
@@ -38,7 +38,7 @@ App.ApplicationController = Em.Controller.extend(App.LoadingSpinnerMixin, App.Al
 		}
 	}),
 
-	init: function () {
+	init() {
```
Comment (Warkot) [inline front/scripts/main/controllers/ApplicationController.ts]: Missing return type.

File: front/scripts/main/controllers/ApplicationController.ts [hunk 10]
```diff
@@ -92,7 +92,7 @@ App.ApplicationController = Em.Controller.extend(App.LoadingSpinnerMixin, App.Al
 		 *
 		 * @param target
 		 */
-		handleLink: function (target: HTMLAnchorElement) {
+		handleLink(target: HTMLAnchorElement) {
```
Comment (Warkot) [inline front/scripts/main/controllers/ApplicationController.ts]: Missing return type.

File: front/scripts/main/controllers/ApplicationController.ts [hunk 11]
```diff
@@ -100,7 +100,7 @@ App.ApplicationController = Em.Controller.extend(App.LoadingSpinnerMixin, App.Al
 		 * @desc Handles query params that should open a lightbox.
 		 * If you add another param to the app you should modify this function.
 		 */
-		handleLightbox: function () {
+		handleLightbox() {
```
Comment (Warkot) [inline front/scripts/main/controllers/ApplicationController.ts]: Missing return type.

File: front/scripts/main/controllers/ApplicationController.ts [hunk 12]
```diff
@@ -114,7 +114,7 @@ App.ApplicationController = Em.Controller.extend(App.LoadingSpinnerMixin, App.Al
 		/**
 		 * @desc Bubbles up to ApplicationRoute
 		 */
-		loadRandomArticle: function () {
+		loadRandomArticle() {
```
Comment (Warkot) [inline front/scripts/main/controllers/ApplicationController.ts]: Missing return type.

File: front/scripts/main/controllers/DiscussionPostController.ts [hunk 13]
```diff
@@ -17,7 +17,7 @@ App.DiscussionPostController = Em.Controller.extend({
 	}),
 
 	actions: {
-		expand: function () {
+		expand() {
```
Comment (Warkot) [inline front/scripts/main/controllers/DiscussionPostController.ts]: Missing return type.

File: front/scripts/main/mixins/LoadingSpinnerMixin.ts [hunk 14]
```diff
@@ -12,7 +12,7 @@ App.LoadingSpinnerMixin = Em.Mixin.create({
 	 * perceived speed of applications is better
 	 * if not, small delay will be almost unnoticeable
 	 */
-	showLoader: function () {
+	showLoader() {
```
Comment (Warkot) [inline front/scripts/main/mixins/LoadingSpinnerMixin.ts]: Missing return type.

File: front/scripts/main/mixins/LoadingSpinnerMixin.ts [hunk 15]
```diff
@@ -20,7 +20,7 @@ App.LoadingSpinnerMixin = Em.Mixin.create({
 		}
 	},
 
-	hideLoader: function () {
+	hideLoader() {
```
Comment (Warkot) [inline front/scripts/main/mixins/LoadingSpinnerMixin.ts]: Missing return type.

File: front/scripts/main/mixins/VisibilityStateManager.ts [hunk 16]
```diff
@@ -33,7 +33,7 @@ App.VisibilityStateManager = Em.Object.create({
 	/**
 	 * @desc runs a loop over this.components and check if they are visible
 	 */
-	check: function () {
+	check() {
```
Comment (Warkot) [inline front/scripts/main/mixins/VisibilityStateManager.ts]: Missing return type.

File: front/scripts/main/mixins/VisibilityStateManager.ts [hunk 17]
```diff
@@ -58,7 +58,7 @@ App.VisibilityStateManager = Em.Object.create({
 		}
 	},
 
-	checkDebounced: function () {
+	checkDebounced() {
```
Comment (Warkot) [inline front/scripts/main/mixins/VisibilityStateManager.ts]: Missing return type.

File: front/scripts/main/mixins/VisibilityStateManager.ts [hunk 18]
```diff
@@ -66,7 +66,7 @@ App.VisibilityStateManager = Em.Object.create({
 	 * @desc adds component to components array and initializes scroll listener
 	 * @param component
 	 */
-	add: function (component: Em.Component) {
+	add(component: Em.Component) {
```
Comment (Warkot) [inline front/scripts/main/mixins/VisibilityStateManager.ts]: Missing return type.

File: front/scripts/main/mixins/VisibilityStateManager.ts [hunk 19]
```diff
@@ -81,7 +81,7 @@ App.VisibilityStateManager = Em.Object.create({
 	/**
 	 * @desc resets state, used in ArticleController on a page change
 	 */
-	reset: function () {
+	reset() {
```
Comment (Warkot) [inline front/scripts/main/mixins/VisibilityStateManager.ts]: Missing return type.

File: front/scripts/main/mixins/VisibleMixin.ts [hunk 20]
```diff
@@ -8,7 +8,7 @@
 'use strict';
 
 App.VisibleMixin = Em.Mixin.create({
-	init: function () {
+	init() {
```
Comment (Warkot) [inline front/scripts/main/mixins/VisibleMixin.ts]: Missing return type.

File: front/scripts/main/models/ArticleCommentsModel.ts [hunk 21]
```diff
@@ -37,7 +37,7 @@ App.ArticleCommentsModel = Em.Object.extend({
 		});
 	}),
 
-	url: function (articleId: number, page: number = 0) {
+	url(articleId: number, page: number = 0) {
```
Comment (Warkot) [inline front/scripts/main/models/ArticleCommentsModel.ts]: Missing return type.

**Comment by Warkot** [comment]
I've heard from @rogatty, that return types are needed for future js docs for ES6 as we will be moving away from TypeScript and `void` in this case will be ignored. So you might ignore it too. But be aware that there are some places here that need a return type, as the respective functions actually return something.

**Comment by bkoval** [comment]
Let's add return types here so we don't create any more merge conflicts with the second refactor.

File: front/scripts/main/components/ArticleCommentComponent.ts [hunk 22]
```diff
@@ -41,7 +41,7 @@ App.ArticleCommentComponent = Em.Component.extend({
 	}),
 
 	actions: {
-		toggleExpand: function () {
+		toggleExpand() {
```
Comment (Warkot) [inline front/scripts/main/components/ArticleCommentComponent.ts]: Missing return type. :hand: :cop:

File: front/scripts/main/components/ForumWrapperComponent.ts [hunk 23]
```diff
@@ -28,16 +28,16 @@ App.ForumWrapperComponent = Em.Component.extend(App.LoadingSpinnerMixin, App.Dis
 		}
 	},
 
-	hasMore: function() {
+	hasMore(): boolean {
 		return this.totalPosts > this.postsDisplayed;
 	},
 
-	pageLoaded: Ember.observer('postsDisplayed', function() {
+	pageLoaded: Ember.observer('postsDisplayed', function () {
```
Comment (Warkot) [inline front/scripts/main/components/ForumWrapperComponent.ts]: Missing return type. :dart:

**Comment by Warkot** [comment]
LGTM

**Comment by hakubo** [comment]
Nice! Thanks!

**Comment by rogatty** [comment]
LGTM

**Comment by lizlux** [comment]
:+1: thanks!

**Pull request merged**